### PR TITLE
Test new integration tests. No new release. 

### PR DESCRIPTION
**Pull request by ablaom**
This follows up #1079 which did not actual trigger integration tests because the source of the PR was not `dev`. 

This PR still needs:

- [x] #1084

### CODE REVIEW

**Comment by ablaom** [comment]
Okay, my attempt [here](https://github.com/alan-turing-institute/MLJ.jl/pull/1079/files) in ci.yml to change the `MLJ_TEST_INTEGRATION` flag does not seem to have worked (julia 1 test is on 6mins, we are expected ~60min). 

@DilumAluthge Are you able to look over my ci.yml logic? This is not a strong point of mine.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -18,13 +18,18 @@ jobs:
...
           - '1' # automatically expands to the latest stable 1.x release of Julia.
         os:
           - ubuntu-latest
         arch:
           - x64
     steps:
+      - name: Set integration test flag
+        run: |
+          julia -e '
+            ENV["MLJ_TEST_INTEGRATION"]="true"'
```
Comment (DilumAluthge) [inline .github/workflows/ci.yml:30]:
```suggestion
            open("ENV["GITHUB_ENV"]", "a") do io
                name = "MLJ_TEST_INTEGRATION"
                value = "true"
                line = "$(name)=$(value)"
                println(io, line)
            end
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -26,9 +26,7 @@ jobs:
     steps:
       - name: Set integration test flag
         run: |
-          julia -e '
-            ENV["MLJ_TEST_INTEGRATION"]="true"'
-
+          echo "TEST_MLJ_INTEGRATION=true" >> $GITHUB_ENV
```
Comment (DilumAluthge) [inline .github/workflows/ci.yml:29]:
```suggestion
          echo "TEST_MLJ_INTEGRATION=true" >> ${GITHUB_ENV:?}
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ablaom) [reply .github/workflows/ci.yml]: Tried in https://github.com/alan-turing-institute/MLJ.jl/pull/1080/commits/fc139f20028e33543204c3dc5dd50451e634b3a5 but also not seeming to work.

**Comment by DilumAluthge** [comment]
Let me take a closer look. Something seems to not be working.

**Comment by DilumAluthge** [comment]
https://github.com/alan-turing-institute/MLJ.jl/pull/1081

**Comment by ablaom** [comment]
Well, this is weird:

Screen Shot 2024-01-15 at 8 41 38 AM

**Comment by DilumAluthge** [comment]
There are two jobs: push and pull_request. The screenshot you posted is from a push job, in which the various PR-related variables won't be defined. Those variables will be defined in the pull_request job.

**Comment by ablaom** [comment]
Go it! Yay, it's working!!

Thanks @DilumAluthge ❤️

**Comment by ablaom** [comment]
Hooray 🎉 . MLJ integration tests are now, well... integrated.

**Pull request merged**